shiny/workflowPlot/server.R

```diff
@@ -3,99 +3,198 @@ library(PEcAn.DB)
+    # Ideally the get_workflow_ids function (line 137) in db/R/query.dplyr.R should take a flag to check
+    # if we want to load all workflow ids.
+    # get_workflow_id function from query.dplyr.R
+    all_ids <- get_workflow_ids(bety, session,all.ids=TRUE)
+    updateSelectizeInput(session, "all_workflow_id", choices=all_ids)
   })
-
-  # update variables if user changes run
-  var_names <- reactive({
-      run_ids <- get_run_ids(bety, workflow_id())
-      var_names <- get_var_names(bety, workflow_id(), run_ids[1])
-      return(var_names)
+  # Update all run ids
+  all_run_ids <- reactive({
+    # Retrieves all run ids for seleted workflow ids
+    # Returns ('workflow ',w_id,', run ',r_id)
+    req(input$all_workflow_id)
+    w_ids <- input$all_workflow_id
+    # Will return a list
+    run_id_list <- c()
+    for(w_id in w_ids){
+    # For all the workflow ids  
+      r_ids <- get_run_ids(bety, w_id)
+      for(r_id in r_ids){
+        # Each workflow id can have more than one run ids
+        # ',' as a separator between workflow id and run id
+        list_item <- paste0('workflow ',w_id,', run ',r_id)
+        run_id_list <- c(run_id_list,list_item)
+      }
+    }
+    return(run_id_list)
   })
+  # Update all run_ids ('workflow ',w_id,', run ',r_id)
   observe({
-    updateSelectizeInput(session, "variable_name", choices=var_names())
+    updateSelectizeInput(session, "all_run_id", choices=all_run_ids())
   })
-
+  return_DF_from_run_ID <- function(diff_ids){
+    # Called by the function parse_ids_from_input_runID
+    # which is a wrapper of this function
+    # Returns a DF for a particular run_id
+    split_string <- strsplit(diff_ids,',')[[1]]
+    # Workflow id is the first element. Trim leading and ending white spaces. Split by space now
+    wID <- as.numeric(strsplit(trimws(split_string[1],which = c("both")),' ')[[1]][2])
+    # Run id is the second element
+    runID <- as.numeric(strsplit(trimws(split_string[2],which = c("both")),' ')[[1]][2])
+    return(data.frame(wID,runID))
+  }
+  # Wrapper over return_DF_from_run_ID
+  # @param list of multiple run ids
+  # run_id_string: ('workflow' workflow_ID, 'run' run_id)
+  # @return Data Frame of workflow and run ids
+  parse_ids_from_input_runID <- function(run_id_list){
+    globalDF <- data.frame()
+    for(w_run_id in run_id_list){
+      globalDF <- rbind(globalDF,return_DF_from_run_ID(w_run_id))
+    }
+    return(globalDF)
+  }  
+  # Fetches variable names from DB
+  # @param workflow_id and run_id
+  # @return List of variable names
+  var_names_all <- function(workflow_id, run_id){
+    # Get variables for a particular workflow and run id
+    var_names <- get_var_names(bety, workflow_id, run_id)
+    # Remove variables which should not be shown to the user
+    removeVarNames <- c('Year','FracJulianDay')
+    var_names <- var_names[!var_names %in% removeVarNames]
+    return(var_names)
+  }
+  # Update variable names  
```

Comment:
Why do we want this in db package ([get_var_names](https://github.com/PecanProject/pecan/blob/develop/db/R/query.dplyr.R#L183) is already there)? Here I am just subsetting to remove var_names related to 'Year' and 'FracJulianDay' which are not used for plotting.